### PR TITLE
Add edit share and tap gesture in profile pic viewer

### PR DESCRIPTION
**Pull request by scoder1747**
## Summary
This PR adds Edit and Share buttons and includes tap gesture in the Profile Pic Image Viewer page. Edit button navigates to the current Edit profile page. The Share button here is unique in that it allows direct image file (not nostr.build or any link) to be shared over Share Sheet (uses sharelink). Tap gesture hides/unhides buttons over tapping action on the page canonical to the Image Carousel viewer.

### Edit navigation:

https://github.com/user-attachments/assets/e651e042-f36f-408c-944f-b421f39ed66b

### Sharing the image directly:

Screenshot 2024-10-23 at 11 11 26 AM

Screenshot 2024-10-23 at 11 11 26 AM

### Tap gesture:

https://github.com/user-attachments/assets/855ef09a-bf92-4a0b-ac71-c9f1d90a313a

_[Please provide a summary of the changes in this PR.]_

## Checklist

- [ ] I have read (or I am familiar with) the [Contribution Guidelines](../docs/CONTRIBUTING.md)
- [x] I have tested the changes in this PR
- [x] My PR is either small, or I have split it into smaller logical commits that are easier to review
- [x] I have added the signoff line to all my commits. See [Signing off your work](../docs/CONTRIBUTING.md#sign-your-work---the-developers-certificate-of-origin)
- [ ] I have added appropriate changelog entries for the changes in this PR. See [Adding changelog entries](../docs/CONTRIBUTING.md#add-changelog-changed-changelog-fixed-etc)
    - [ ] I do not need to add a changelog entry. Reason: _[Please provide a reason]_
- [x] I have added appropriate `Closes:` or `Fixes:` tags in the commit messages wherever applicable, or made sure those are not needed. See [Submitting patches](https://github.com/damus-io/damus/blob/master/docs/CONTRIBUTING.md#submitting-patches)

### CODE REVIEW

File: damus/Views/Images/ProfilePicImageView.swift [hunk 1]
```diff
@@ -93,18 +96,57 @@ struct ProfilePicImageView: View {
...
             .modifier(SwipeToDismissModifier(minDistance: 50, onDismiss: {
                 presentationMode.wrappedValue.dismiss()
             }))
         }
-        .overlay(NavDismissBarView(navDismissBarContainer: .profilePicImageView), alignment: .top)
+        .overlay(
+            Group {
+                if showMenu {
+                    HStack {
+                        NavDismissBarView()
```
Comment (danieldaquino) [inline damus/Views/Images/ProfilePicImageView.swift:115]: This line is failing to compile with the following error message: 

```
Missing argument for parameter 'navDismissBarContainer' in call
```

I believe it is due to a recent parallel change on `NavDismissBarView`, that now requires an argument for this call

File: damus/Views/Images/ProfilePicImageView.swift [hunk 2]
```diff
@@ -93,18 +96,57 @@ struct ProfilePicImageView: View {
...
+                        if let image = image {
+                            ShareLink(item: Image(uiImage: image),
+                                      preview: SharePreview("Damus Profile", image: Image(uiImage: image))) {
```
Comment (danieldaquino) [inline damus/Views/Images/ProfilePicImageView.swift:118]: We should localize "Damus Profile" by using `NSLocalizedString`

**Review by danieldaquino — CHANGES_REQUESTED** [review]
Thank you @scoder1747 for your PR! While trying to review and test it, I have encountered a build error, which I believe is because the merge brought some changes that affect this code. Can you please check?

**Comment by scoder1747** [comment]
Thanks @danieldaquino! I’ve addressed your comments and made the updates.

**Comment by danieldaquino** [comment]
Thank you @scoder1747! Changes look good and are working on my end as well!

**I squashed the commits into one and manually pushed to `master`: 6f044553501ff0141e8425ce8aa260f73e101913**

A few extra comments and tips in terms of git commits:
1. I noticed one changelog entry called "_Address PR feedback_". Don't worry about adding a changelog entry for each commit, they are only needed for user-facing changes
2. If you need to tweak your PR, sometimes it's ok to add more commits to your branch, but it can be helpful to learn how to use `git rebase --interactive` to edit past commits in your PR and keep a cleaner commit history.

Thank you!

**Pull request closed without merging**